cmake/Modules/FindNCCL.cmake

```diff
@@ -31,7 +31,13 @@
 # install NCCL in the same location as the CUDA toolkit.
 # See https://github.com/caffe2/caffe2/issues/1601
 
-set(NCCL_ROOT_DIR "" CACHE PATH "Folder contains NVIDIA NCCL")
+set(NCCL_ROOT $ENV{NCCL_ROOT} CACHE PATH "Folder contains NVIDIA NCCL")
```

Comment:
> You can also set(ENV{variable_name} value) and get $ENV{variable_name} environment variables, though it is generally a very good idea to avoid them.

https://cliutils.gitlab.io/modern-cmake/chapters/basics/variables.html

Should the env variable checks here be removed? You could just delete these 3 lines.

Reply:
So I am guessing I should remove it here too?

https://github.com/ChaiBapchya/incubator-mxnet/blob/a9abba5a934755813481e6583435298a0c317957/cmake/Modules/FindNVTX.cmake#L18

Reply:
or is it specifically to do with ENV vars. In that case we can set "" to these 3 variables - NCCL_ROOT, NCCL_INCLUDE_DIR and NCCL_LIB_DIR
Since that seems to be the norm

Reply:
@ChaiBapchya let's focus on one thing in a PR. I would change the NVTX_ROOT_DIR in separate PR if needed.

Reply:
@ChaiBapchya Yes, for this PR my concern is only that you are introducing environment variable depencies, which is not standard practice and not needed.
So for this PR, let's just remove the 3 lines.
For a later PR, NVTX can be refactored.

Reply:
@leezu Setting empty string to a cache variable does serve a purpose (declaring a variable so as to check if it is set or not in future)
So if NCCL_ROOT_DIR is not already set then it assigns a default value to it (right now since we have deleted this line, if variable is not set via command line, it won't exist -> hence can't use it in this file)

What do you think?

https://stackoverflow.com/questions/59867283/whats-the-point-of-using-empty-string-to-set-cache-variable-in-cmake/59868538#59868538

Reply:
Yes, I agree with the Stackoverflow answer. In your case, it would "assigns description" to the variable. But as these are deprecated variables I don't think we need to add description.
The standard / supported variable is `NCCL_ROOT` as per https://cmake.org/cmake/help/latest/policy/CMP0074.html